File: Leetcode/arrays/349_IntersectionOfTwoArrays/Solution.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        set<int> record(nums1.begin(), nums1.end());
        set<int> res;
        for (int i = 0; i < nums2.size(); i++)
            if (record.find(nums2[i]) != record.end())
                res.insert(nums2[i]);
        return vector<int>(res.begin(), res.end());
    }
};

File: Leetcode/arrays/349_IntersectionOfTwoArrays/Solution.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        HashSet<Integer> set = new HashSet<>(), set2 = new HashSet<>();
        for(int num: nums1){
            set.add(num);
        }
        for(int num: nums2){
            if(set.contains(num)){
                set2.add(num);
            }
        }
        int[] res = new int[set2.size()];
        int i = 0;
        for(int num: set2){
            res[i++] = num;
        }
        return res;
    }
}
